### PR TITLE
feature(xjx): filter topic before parsing the message

### PR DESCRIPTION
**Pull request by sailxjx**
I added a `topic` to the beginning of the message, so we don't need to call pickle.loads when the current process is not subscribed to the topic.

### CODE REVIEW

File: ding/framework/parallel.py [hunk 1]
```diff
@@ -253,24 +253,35 @@ def send_rpc(self, func_name: str, *args, **kwargs) -> None:
...
 
-    def recv_rpc(self, msg: bytes):
+    def recv_rpc(self, msg: bytes) -> None:
+        """
+        Overview:
+            Recv and parse payload from other processes, and call local functions.
+        Arguments:
+            - msg (:obj:`bytes`): Recevied message.
+        """
+        topic, payload = msg.split(b"::", maxsplit=1)
```
Comment (PaParaZz1) [inline ding/framework/parallel.py:273]: add comment about why use topic (such as a link to corresponding communication topology graph)

File: ding/framework/parallel.py [hunk 2]
```diff
@@ -253,24 +253,35 @@ def send_rpc(self, func_name: str, *args, **kwargs) -> None:
             - fn_name (:obj:`str`): Function name.
         """
         if self.is_active:
-            payload = {"f": func_name, "a": args, "k": kwargs}
+            topic = func_name + "::"
```
Comment (PaParaZz1) [inline ding/framework/parallel.py:256]: whether more identifiers need to be added for avoiding misuse. (like `topic::`)

File: ding/framework/parallel.py [hunk 3]
```diff
@@ -253,24 +253,35 @@ def send_rpc(self, func_name: str, *args, **kwargs) -> None:
...
+        topic, payload = msg.split(b"::", maxsplit=1)
+        func_name = topic.decode()
+        if func_name not in self._rpc:
+            logging.warning("There was no function named {} in rpc table".format(func_name))
+            return
         try:
-            payload = pickle.loads(msg)
+            payload = pickle.loads(payload)
         except Exception as e:
             logging.warning("Error when unpacking message on node {}, msg: {}".format(self._bind_addr, e))
```
Comment (PaParaZz1) [inline ding/framework/parallel.py:281]: why use `logging.warning` here, `logging.error` can be better?

**Review by PaParaZz1 — APPROVED** [review]

File: ding/framework/parallel.py (hunk 3 above)
Comment (sailxjx) [reply ding/framework/parallel.py]: done

File: ding/framework/parallel.py (hunk 2 above)
Comment (sailxjx) [reply ding/framework/parallel.py:256]: I only split the first `::` when receiving the message, there is no need to add additional letters.

**Comment by sailxjx** [comment]
Move to another branch

**Pull request closed without merging**